api/src/main/java/org/apache/iceberg/exceptions/NoCommitSucceedException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.exceptions;
+
+import com.google.errorprone.annotations.FormatMethod;
+
+/**
+ * Exception raised when no commit succeed.
+ * <p>
+ * For example, this is thrown when rewriting data files partial commit is enabled,
+ * but no commit succeed at the end.
+ */
+public class NoCommitSucceedException extends RuntimeException {
```

Comment:
I'd rather not add this. I think it is going to cause confusion because we would have both `CommitFailedException` and `NoCommitSucceededException`. If you want to define this within the action, like `RewriteDataFiles.NoSuccessfulCommit` then that's okay. But we shouldn't add it to the general API.

Reply:
@rdblue sounds great! I'll I will define this exception class in `RewriteDataFiles`.